drizzlepac/runastrodriz.py

```diff
@@ -1848,7 +1909,14 @@ def update_active_wcs(filename, wcsname, logfile=None):
         msg = "No need to update active WCS solution of {} for {} as it is already the active solution.\n".format(wcsname, filename)
         print(msg)
         update_msg += msg
-
+    try:
+        idcscale = fits.getval(filename, ext=1, keyword='IDCSCALE')
+    except KeyError:
+        msg += "Headerlet {} was missing IDCSCALE keyword".format(wcsname)
+        print(msg)
+        update_msg += msg
+        _update_idcscale(filename)
```

Comment:
Wouldn't you need to try getting `idcscale` again using the same code as on line 1912? (`idcscale = fits.getval(filename, ext=1, keyword='IDCSCALE')`)

Reply:
I'm not sure I understand. Line 1913 checks the science header which was just updated by a headerlet. `_update_idcscale` iterates over all headerlets in the file to retrieve an `idcscale` for the same `idctab` and if it doesn't find any, computes one from the CD matrix.
What am I missing?